### PR TITLE
Upgrade to JTS 1.17.0.

### PR DESCRIPTION
**Pull request by jnh5y**
Signed-off-by: Jim Hughes <jnh5y@ccri.com>

### CODE REVIEW

**Comment by dsmiley** [comment]
Thanks!  Can you update CHANGES.md as well please?  In like-kind to past entries.

**Comment by jnh5y** [comment]
@dsmiley of course.  Are the numbers, the PR number or the issue number?  (I assumed the former.)

Any more that I can say for the update?

File: CHANGES.md [hunk 1]
```diff
@@ -2,6 +2,10 @@
 
 DATE: _unreleased_
 
+* \#188: Upgraded to JTS 1.17.0.  This JTS release has a small [API change](https://github.com/locationtech/jts/blob/master/doc/JTS_Version_History.md#api-changes).  
+  LocationTech JTS has updated its license and is now dual-licensed as EPL 2.0 and EDL 1.0 (a BSD style license).
```
Comment (dsmiley) [inline CHANGES.md:6]: JTS's license has not changed for 1.17 so why mention it here?

Comment (jnh5y) [reply CHANGES.md]: I thought it had changed....  I've removed the note (it sounds like you didn't see the change in an obvious place).

**Comment by dsmiley** [comment]
JTS 1.17 included a change to upgrade Java 1.7 to 1.8 -- https://github.com/locationtech/jts/releases/tag/1.17.0 and this means Spatial4j's builds must now require at least 1.8 as well, but it can keep javac source & target options to 1.7. Users can continue to run Spatial4j with Java 1.7 but will need to upgrade to use the JTS we specify if they want to use JTS (it's an optional dependency). Strangely, TravisCl hasn't been pinging me about the failure but I did seek it out: 
https://travis-ci.org/github/locationtech/spatial4j/builds/706041451 
I wish it showed up here in the PR.  Hmm; something to look in to.

**Comment by jnh5y** [comment]
Sorry for that oversight on my part.  I didn't realized that Spatial4J was still supporting Java 7 and that JTS had moved up to requiring Java 8 in 1.17.0.  

I don't see the results from the Git WebHook with Travis CI here.  Maybe those need to configured differently?  I know for GeoMesa that the GitHub webhooks will show green check or red X's for a number of criteria.  (Let me know if I can help with any of that.)

**Comment by dsmiley** [comment]
Yes please, I would appreciate help hooking up Travis CI to the PR.  Do I just need to RTFM (URL please)?

**Comment by jnh5y** [comment]
We'll likely have to email the Eclipse webmaster to ask about some GitHub settings.  

After thinking about it, would it be worth considering GitHub Actions instead?  (If you are willing to try them, I may be able to find a few minutes to gin up an example of GH Actions that'd run the builds you are using Travis for.)

If we go that route, is Java 7 support still important to you?  (If so, that's fine.  That said, I'd love to hear a reason why.)

**Comment by dsmiley** [comment]
No Java 7 is too old, but I still -source & -target at the complication step to 7.  Note that since I added in JDK 11 to Travis very recently, there seems to be some test maintenance for me to do in order to make tests work in JDK 11. Work for me to do soon.

**Pull request merged**